### ISSUE TITLE
[Interface-name] Never prefix return error for I18nInterface

### ISSUE THREAD

**Issue by VincentLanglet** [palantir/tslint#4476]
### Bug Report

- __TSLint version__: 5.12.1
- __TypeScript version__: 3.2.4
- __Running TSLint via__: CLI / PhpStorm

#### TypeScript code being linted

```ts
interface I18nInterface {
  foo: () => void;
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:all"],
  "rules": {
    "interface-name": [true, "never-prefix"]
  }
}
```

#### Actual behavior
Got error "Interface must not have an I prefix

#### Expected behavior
No error, I18n should be an allowed word.
Don't know if we have to allow ^I\d+ interfaces or just ^I18n ones.

**Comment by adidahiya** [comment]
looks like a bug. there's already a special case in the rule for `IDB`, might as well add one for `I18`

**Comment by adidahiya** [comment]
Fixed by #4486
